modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheDatabaseSharedManager.java

```diff
@@ -1848,17 +1855,54 @@ private boolean safeToUpdatePageMemories() {
             }
         }
 
-        if (log.isInfoEnabled()) {
-            log.info("Following partitions were reserved for potential history rebalance [" +
-                grpPartsWithCnts.entrySet().stream().map(entry ->
-                    "grpId=" + entry.getKey() +
-                    ", grpName=" + cctx.cache().cacheGroupDescriptor(entry.getKey()).groupName() +
-                    ", parts=" + S.compact(entry.getValue().keySet())).collect(Collectors.joining(", ")) + ']');
-        }
+        if (log.isInfoEnabled() && !F.isEmpty(earliestValidCheckpoints))
+            printReservationToLog(earliestValidCheckpoints);
 
         return grpPartsWithCnts;
     }
 
+    /**
+     * Prints detail information about caches which were not reserved
+     * and reservation depth for the caches which have WAL history enough.
+     *
+     * @param earliestValidCheckpoints Map contains information about caches' reservation.
+     */
+    private void printReservationToLog(
+        Map<Integer, T2<String, Map<Integer, CheckpointEntry>>> earliestValidCheckpoints) {
+        Map</*grpId*/Integer, T2</*reason*/String, Map</*partId*/Integer, CheckpointEntry>>> notReservedCaches =
+            earliestValidCheckpoints.entrySet().stream()
+                .filter(entry -> entry.getValue().get2() == null)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        if (!F.isEmpty(notReservedCaches)) {
+            log.info("Following caches were not reserved [" +
+                notReservedCaches.entrySet().stream()
+                    .map(entry -> "[grpId=" + entry.getKey() +
+                        ", grpName=" + cctx.cache().cacheGroup(entry.getKey()).cacheOrGroupName() +
+                        ", reason=" + entry.getValue().get1() + ']')
+                    .collect(Collectors.joining(", ")) + ']');
+        }
+
+        Map</*grpId*/Integer, T2</*reason*/String, Map</*partId*/Integer, CheckpointEntry>>> reservedCaches =
+            earliestValidCheckpoints.entrySet().stream()
+                .filter(entry -> entry.getValue().get2() != null)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        if (!F.isEmpty(reservedCaches)) {
+            log.info("Caches were reserved in WAL not deeper than last checkpoint by specified reason [" +
```

Comment:
"Reserved cache groups with first reserved checkpoint IDs and reasons why previous checkpoint was inapplicable:"

Reply:
Done.